test/integration/base_integration_test.cc

```diff
@@ -694,4 +698,45 @@ AssertionResult BaseIntegrationTest::compareDeltaDiscoveryRequest(
   return AssertionSuccess();
 }
 
+void BaseIntegrationTest::checkForMissingTagExtractionRules() {
+  BufferingStreamDecoderPtr response = IntegrationUtil::makeSingleRequest(
+      test_server_->adminAddress(), "GET", "/config_dump", "", Http::CodecType::HTTP1);
+  ASSERT_TRUE(response->complete());
+  Json::ObjectSharedPtr json = Json::Factory::loadFromString(response->body());
+  std::vector<std::string> stat_prefixes;
+  Json::ObjectCallback find_stat_prefix = [&](const std::string& name,
+                                              const Json::Object& root) -> bool {
+    if (name == "stat_prefix") {
+      auto prefix = root.asString();
+      if (!prefix.empty()) {
+        stat_prefixes.push_back(prefix);
+      }
+    } else if (root.isObject()) {
+      root.iterate(find_stat_prefix);
+    } else if (root.isArray()) {
+      std::vector<Json::ObjectSharedPtr> elements = root.asObjectArray();
+      for (const auto& element : elements) {
+        find_stat_prefix("", *element);
+      }
+    }
+    return true;
+  };
+  find_stat_prefix("", *json);
+  ENVOY_LOG_MISC(debug, "discovered stat_prefixes {}", stat_prefixes);
+
+  auto check_metric_array = [&](auto vec) {
+    for (const auto& metric : vec) {
+      const std::string tag_extracted_name = metric->tagExtractedName();
+      for (const std::string& stat_prefix : stat_prefixes) {
+        EXPECT_EQ(tag_extracted_name.find(stat_prefix), std::string::npos)
```

Comment:
I don't understand how just doing a total substring search for the prefixes in the tag-extracted name tells you whether the tag-extraction regexes are complete.

Reply:
The idea is that if the `stat_prefix` is in the tag-extracted name, then it wasn't extracted, which I believe is always an error (although there are many such cases in the codebase right now). 

Or at least that's how I use `stat_prefix`. Maybe others use it completely differently, and this isn't correct. I'd like to hear opinions on this.

Reply:
Also, this assumes that the configured `stat_prefix` in the integration tests is a value that is unique enough that it isn't  a substring of any other part of the stat name.

Reply:
Another approach to this test, that may be more robust, is to instead look at only the extracted tags, and verify that the value configured for `stat_prefix` is in one of the extracted tags, and ignore the tagExtractedName. That may more clearly express what I'm trying to accomplish.